pkg/controller/storagecluster/storageclasses.go

```diff
@@ -162,6 +136,38 @@ func (r *ReconcileStorageCluster) newStorageClasses(initData *ocsv1.StorageClust
 				"csi.storage.k8s.io/controller-expand-secret-namespace": initData.Namespace,
 			},
 		},
+		reconcileStrategy: ReconcileStrategy(managementSpec.ReconcileStrategy),
+		disable: managementSpec.DisableStorageClass,
+	}
+}
+
+func (r *ReconcileStorageCluster) newOBCStorageClass(initData *ocsv1.StorageCluster) *StorageClassConfiguration {
+	reclaimPolicy := corev1.PersistentVolumeReclaimDelete
+	managementSpec := initData.Spec.ManagedResources.CephObjectStores
+	return &StorageClassConfiguration{
+		storageClass: &storagev1.StorageClass{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: generateNameForCephRgwSC(initData),
+			},
+			Provisioner:   fmt.Sprintf("%s.ceph.rook.io/bucket", initData.Namespace),
+			ReclaimPolicy: &reclaimPolicy,
+			Parameters: map[string]string{
+				"objectStoreNamespace": initData.Namespace,
+				"region":               "us-east-1",
+				"objectStoreName":      generateNameForCephObjectStore(initData),
+			},
+		},
+		reconcileStrategy: ReconcileStrategy(managementSpec.ReconcileStrategy),
+		disable: managementSpec.DisableStorageClass,
+	}
+}
+
+// newStorageClasses returns the StorageClass instances that should be created
+// on first run.
+func (r *ReconcileStorageCluster) newStorageClasses(initData *ocsv1.StorageCluster) ([]*StorageClassConfiguration, error) {
+	ret := []*StorageClassConfiguration{
+		r.newFilesystemStorageClass(initData),
+		r.newBlockPoolStorageClass(initData),
```

Comment:
These calls don't need a `ReconcileStorageCluster` receiver as they neither access it nor mutate it.